### PR TITLE
Use JAVA_HOME, if available, to find keytool

### LINKED ISSUE #3903: Keytool command used in generateKeyStore doesnt work in windows
The `keytool` command is used for UAA server option in `generateKeyStore` method of `generator-base.js` its not cross platform as the command needs to be executed from the jdk/jre bin folder. not sure if this is applicable for mac and linux @jhipster/developers can anyone check

@sdoxsee since this is from you could you check?

### PR DESCRIPTION
**Pull request by sdoxsee**
fix #3903

### CODE REVIEW

File: generators/generator-base.js [hunk 1]
```diff
@@ -1164,7 +1164,12 @@ Generator.prototype.generateKeyStore = function() {
     } else {
         shelljs.mkdir('-p', SERVER_MAIN_RES_DIR);
         var parent = this;
-        shelljs.exec('keytool '+
+        var javaHome = shelljs.env['JAVA_HOME'];
+        var keytoolPath = '';
+        if (javaHome) {
+            keytoolPath = javaHome + '/bin/';
+        }
+        shelljs.exec(keytoolPath + 'keytool '+
```
Comment (deepu105) [inline generators/generator-base.js]: @sdoxsee this still doesnt work in windows, the below works. can you check if its ok for osx?
`shelljs.exec('"' + keytoolPath + 'keytool" '+`

**Comment by sdoxsee** [comment]
@deepu105 yes. your change works here on OSX.

**Comment by deepu105** [comment]
ok ill change it


Thanks & Regards,
Deepu

**Comment by sdoxsee** [comment]
Thanks!

**Pull request merged**